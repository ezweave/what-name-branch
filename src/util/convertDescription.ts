import { flow, toLower } from 'lodash';
import { replace as replaceFP } from 'lodash/fp';
import { trim } from './trim';

interface ConvertDescription {
  (description: string): string,
}

export const convertDescription: ConvertDescription = flow(
  trim,
  toLower,
  replaceFP(/\s+/g, '-'),
  replaceFP(/[^\w-_]+/g, ''),
);
