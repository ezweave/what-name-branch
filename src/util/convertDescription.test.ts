import { convertDescription } from './convertDescription';

describe(convertDescription, () => {
  it('converts any description to a git/file safe name', () => {
    expect(
      convertDescription('the S3 notifications are too notify!  how fix? #!@##$'),
    ).toMatchSnapshot();
  });
  it('removes unsafe characters', () => {
    expect(
      convertDescription('issue-name$#!@%$*&))"\''),
    ).toEqual('issue-name');
  });
});
