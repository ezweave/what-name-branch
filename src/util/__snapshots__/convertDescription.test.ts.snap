// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` converts any description to a git/file safe name 1`] = `"the-s3-notifications-are-too-notify-how-fix-"`;
